src/app/pages/subjects/book-viewer/book-viewer.js

```diff
@@ -4,46 +4,46 @@ import {ActiveElementContextProvider} from '~/contexts/active-element';
 import useSubjectsContext from '../context';
 import BookCover from './book';
 import useSubjectCategoryContext from '~/contexts/subject-category';
+import cn from 'classnames';
 
 function useCategorizedBooks() {
     const {books} = useSubjectsContext();
     const categories = useSubjectCategoryContext();
-    const result = {};
-    const addLabels = () => {
-        for (const category of categories) {
-            if (result[category.cms]) {
-                result[category.cms].label = category.html;
-            }
-        }
-    };
 
-    for (const book of books) {
-        book.subjects.forEach((cmsCategory) => {
-            if (!(cmsCategory in result)) {
-                result[cmsCategory] = [];
-            }
-            if (!result[cmsCategory].includes(book)) {
-                result[cmsCategory].push(book);
+    return React.useMemo(
+        () => {
+            const result = {};
+
+            for (const book of books) {
+                book.subjects.forEach((cmsCategory) => {
+                    if (!(cmsCategory in result)) {
+                        result[cmsCategory] = [];
+                    }
+                    if (!result[cmsCategory].includes(book)) {
+                        result[cmsCategory].push(book);
+                    }
+                });
             }
-        });
-    }
 
-    addLabels();
+            for (const category of categories) {
+                if (result[category.cms]) {
+                    result[category.cms].label = category.html;
+                }
+            }
 
-    return result;
+            return result;
+        },
+        [books, categories]
+    );
 }
 
 function CategorySection({categoryData, categorizedBooks, category}) {
-    const classList = ['book-category'];
+    const hidden = !['view-all', categoryData.value].includes(category);
     const subjectHtml = categoryData.html;
     const books = categorizedBooks[categoryData.cms] || [];
 
-    if (!['view-all', categoryData.value].includes(category)) {
-        classList.push('hidden');
-    }
-
     return (
-        <div className={classList.join(' ')}>
+        <div className={cn('book-category', {hidden})}>
```

Comment:
a tiny thing, but I find it clearer to move the test as close to where it's used as possible.  maybe this would be clearer:
```jsx
<div 
  className={cn('book-category', { 
    hidden: !['view-all', categoryData.value].includes(category),
  })}
>
```

Reply:
I actually prefer the encapsulation on things like this.